jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/MultiPart.java

```diff
@@ -1392,6 +1392,7 @@ private boolean parseContent(Content.Chunk chunk)
             partialBoundaryMatch = boundaryFinder.endsWith(buffer);
             if (partialBoundaryMatch > 0)
             {
+                notifyCRContent();
```

Comment:
See my comments in https://github.com/jetty/jetty.project/pull/11388#discussion_r1490603161 as I don't think this is always correct and is at odds with the comment below.

Reply:
@gregw I think this `notifyCRContent()` is correct.

We may have a CR left over from a previous buffer which has not been emitted as content. And at this point (where we have a partial match with content in this buffer before the match), the saved CR can no longer be part of the boundary, so we must emit it and check again at `buffer.get(sliceLimit - 1)` before we can emit the other content.

I think your comment in the other PR is correct but with the `if (partialBoundaryMatch == buffer.remaining())` block here, we can no longer reach the `notifyCRContent()` if the match starts at the beginning of the current buffer.

So I think the check for `sliceLimit > 0` on line 1411 is now redundant because it is not possible. Shouldn't this have been `sliceLimit > pos` anyway?

Reply:
@lachlan-roberts good point re `sliceLimit > 0`, I removed it.